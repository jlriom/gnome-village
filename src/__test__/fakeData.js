export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
