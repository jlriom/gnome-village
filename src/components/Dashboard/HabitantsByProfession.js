import React from "react";
import PropTypes from "prop-types";
import CategoryItem from "./CategoryItem";

const HabitantsByProfession = ({ habitantsByProfession }) => (
	<ul className='list-group my-4'>
		<li className='list-group-item  list-group-item-secondary'>
			Habitans by profession
		</li>
		{habitantsByProfession.map(profession => (
			<CategoryItem
				key={profession.profession}
				category={profession.profession}
				count={profession.count}></CategoryItem>
		))}
	</ul>
);

HabitantsByProfession.propTypes = {
	habitantsByProfession: PropTypes.array.isRequired
};

export default HabitantsByProfession;
