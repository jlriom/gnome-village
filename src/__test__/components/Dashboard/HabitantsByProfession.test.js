import React from "react";
import HabitantsByProfession from "../../../components/Dashboard/HabitantsByProfession";
import renderer from "react-test-renderer";
import { habitantsByProfession } from "../../fakeData";

describe("<HabitantsByProfession />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <HabitantsByProfession habitantsByProfession={habitantsByProfession} />
        );
        expect(tree).toMatchSnapshot();
    });
});
