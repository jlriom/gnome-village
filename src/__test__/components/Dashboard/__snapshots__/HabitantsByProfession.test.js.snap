// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantsByProfession /> renders ok 1`] = `
<ul
  className="list-group my-4"
>
  <li
    className="list-group-item  list-group-item-secondary"
  >
    Habitans by profession
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Baker
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        214
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Blacksmith
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        180
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Brewer
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        200
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Butcher
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        210
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Carpenter
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        216
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Cook
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        190
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Farmer
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        205
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Gemcutter
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        186
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Leatherworker
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        185
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Marble Carver
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        190
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Mason
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        214
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Mechanic
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        182
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Medic
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        187
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Metalworker
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        192
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Miner
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        201
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Potter
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        202
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Prospector
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        202
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Sculptor
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        182
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Smelter
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        203
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Stonecarver
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        187
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Tailor
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        199
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Tax inspector
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        192
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
         Tinker
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        202
      </span>
    </span>
  </li>
  <li
    className="list-group-item "
  >
    <span
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        Woodcarver
         
      </span>
      <span
        className="badge  badge-pill badge-secondary "
      >
        199
      </span>
    </span>
  </li>
</ul>
`;
